src/engine/sparqlExpressions/LangExpression.h

```diff
@@ -38,7 +38,10 @@ class LangExpressionImpl : public NaryExpression<NaryOperation> {
       return stringPtr->value();
     } else {
       throw std::runtime_error{
-          "use LANG() with ?var as an argument within a Filter-expression"};
+          "Use the LANG() function within a FILTER() expression only with a "
+          "variable as its argument. Valid example: "
+          "FILTER(LANG(?example_var) "
+          "= \"en\")"};
```

Comment:
I don't think this needs to be a hard error anymore.
Can't you just a `std::optional<Variable>` and,
in the cases where there is no variable just don't apply the special language filter implementation (This works now in general (if we ignore issues in the `=` expression for local vocab, which are not your fault, and which I'll have to fix soon).